docs/src/modules/components/MarkdownElement.js

```diff
@@ -773,10 +780,10 @@ const Root = styled('div')(
           },
         },
       },
-      '& a, & a code': {
+      '& a:not(.title-link-to-anchor), & a:not(.title-link-to-anchor) code': {
```

Comment:
I'm reverting this approach, it increased the specificity, leading to overrides other styles: #42735. It seems that we don't need this more complex CSS selector, we can solve the same problem with the CSS injection order.